tests/zfs-tests/tests/functional/casenorm/mixed_create_failure.ksh

```diff
@@ -0,0 +1,56 @@
+#!/bin/ksh -p
+#
+#
+# This file and its contents are supplied under the terms of the
+# Common Development and Distribution License ("CDDL"), version 1.0.
+# You may only use this file in accordance with the terms of version
+# 1.0 of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source.  A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+#
+# Copyright 2018 Nutanix, Inc.  All rights reserved.
+#
+
+. $STF_SUITE/tests/functional/casenorm/casenorm.kshlib
+
+# DESCRIPTION:
+# For the filesystem with casesensitivity=mixed, normalization=none,
+# when multiple files which the same name (differing only in case) created, the
+# number of files is limited to what can fit in a fatzap leaf-block. And beyond
+# that, it fails with ENOSPC. Ensure that the create fails gracefully and does
+#  not hit and assert and panic.
+#
+# STRATEGY:
+# 1. Create files with same name but varying in case.
+#    E.g. 'abcdefghijklmnop', 'Abcdefghijklmnop', 'ABcdefghijklmnop' etc.
+#    The create would fail with ENOSPC.
+
+verify_runnable "global"
+
+function cleanup
+{
+        destroy_testfs
+}
+
+log_onexit cleanup
+log_assert "With mixed mode: ensure create fails with ENOSPC beyond a certain limit"
+
+create_testfs "-o casesensitivity=mixed -o normalization=none"
+
+names='{a,A}{b,B}{c,C}{d,D}{e,E}{f,F}{g,G}{h,H}{i,I}{j,J}{k,K}{l,L}'
+for name in $names; do
+        create_file $name
+	ret=$?
+	log_note "Created file $name status $ret"
+	if (($ret != 0)); then
+		# It is sufficient to test that create fails and does not cause a panic.
+		break;
+	fi
+done
+
+destroy_testfs
+
+log_pass "CM-not-UN FS: lookup succeeds only if using exact name form"
```

Comment:
Please update the message here to reflect what's being tested.

Reply:
oops.. slip up. Fixed it.